backend/models/Contract.js

```diff
@@ -0,0 +1,30 @@
+"use strict";
+module.exports = (sequelize, DataTypes) => {
+  const Contract = sequelize.define(
```

Comment:
Well, having a copy of a table (same as Account) is suboptimal.

Let's discuss if we have leave without it.

To know if there is any contract deployed to the account_id you can query RPC:

```
$ http post https://rpc.nearprotocol.com jsonrpc=2.0 id=dontcare method=query 'params:=["account/metanear-chat",""]'
```

and if the `"code_hash"` is not `"11111111111111111111111111111111"`, then there is a contract, so we can query that information from the frontend on demand. This code hash is not the transaction hash where it was deployed, but the hash of the binary file.

To know the time and the transaction hash of the latest update, you query Actions for `ContractDeployed` action_type joining with the transactions table to filter by `receiver_id`.

Reply:
why do we need to distinguish between accounts and contracts? The distinction is rather superficial and we can choose whether to display anything about contract based on code hash.

Reply:
@kcole16 Do we have someone's opinion/vision on the separation of Accounts and Contracts into separate pages? Would we consider having this as one entity on the Explorer?

Well, just from the user perception, one may have a mental model that there are accounts (users ~= humans) and contracts (programs), and they are not the same, but we, probably, should not encourage this mental model...

Reply:
@bowenwang1996 We aren't distinguishing between accounts and contracts currently. We're just indicating whether or not there is a contract deployed to an account.

@frol Agree we shouldn't encourage a mental model of separation. For now, I don't think the list of contracts is that useful, though when we have contracts matching certain standards (fungible/non-fungible tokens, etc), that might make more sense

Reply:
@icerove So, to sum it up, we don't need separate pages for contracts list and contract details.

@kcole16 I think we should replace the top bar "Contracts" with "Accounts". Agree?

Reply:
Yes, the contract details table should be an additional section on the Account Detail page.

Reply:
@frol  We just refactor most of the SQL query without joining, should we do that here?
>To know the time and the transaction hash of the latest update, you query Actions for ContractDeployed action_type joining with the transactions table to filter by receiver_id.